File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_block(int i);
void print_whitespace(int i);
void print_pyramid(int height);

int main(void)
{
    // get hieght value from user
    int height;
    do
    {
        height = get_int("how height? ");
    }
    while (height < 1 || height > 8);

    // print pyramid
    print_pyramid(height);
}

// print block's in current row
void print_block(int i)
{
    int j;
    for (j = 0; j < i; j++)
    {
        printf("#");
    }
}

// print whitespace's in current row
void print_whitespace(int i)
{
    int j;
    for (j = 0; j < i; j++)
    {
        printf(" ");
    }
}

// print pyramid
void print_pyramid(int height)
{
    int i;
    for (i = 0; i < height; i++)
    {
        print_whitespace(height - (i + 1));

        print_block(i + 1);

        printf("  ");

        print_block(i + 1);

        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
def main():
    while True:
        try:
            # get hieght value from user
            height = int(input("Enter an integer from 1 to 8 inclusive (the height of the half pyramid): "))
            if height > 0 and height < 9:
                print_pyramid(height)
                break
        except ValueError:
            print("Oops!  That was no valid number.  Try again...")


# print block's in current row
def print_block(n):
    for _ in range(n):
        print("#", end="")


# print whitespace's in current row
def print_whitespace(n):
    for _ in range(n):
        print(" ", end="")


# print pyramid
def print_pyramid(height):
    for i in range(height):
        print_whitespace(height - (i + 1))
        print_block(i + 1)
        print_whitespace(2)
        print_block(i + 1)
        print()


main()
